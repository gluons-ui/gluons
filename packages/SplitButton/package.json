{
  "author": "Lucas K Allmon",
  "dependencies": {
    "lodash.clamp": "^4.0.3"
  },
  "description": "Mandala UI Split Button Component",
  "keywords": [
    "react",
    "css",
    "library",
    "components",
    "interface",
    "functional",
    "OOCSS",
    "declarative",
    "ui",
    "user",
    "ux"
  ],
  "license": "MIT",
  "main": "lib/SplitButton.js",
  "module": "es/SplitButton.js",
  "name": "@mandala-ui/split-button",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "prepare": "nwb build-react-component"
  },
  "version": "1.0.2"
}
