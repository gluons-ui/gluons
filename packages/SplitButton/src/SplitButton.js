import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

class SplitButton extends PureComponent {
  render() {
    const {
      baseColor,
      disabled,
      isOn,
      offColor,
      onClick,
      onColor,
      pill,
      radius,
      showText,
    } = this.props;
    return (
      <div className="split-button">
        <div
          className={`b--gray ba
            ${pill ? 'br-pill' : `br${radius}`} flex h2 overflow-hidden w4`}
        >
          <button
            className={`bn w-50
              ${disabled ? 'o-60' : 'pointer'}
              ${isOn ? `bg-${baseColor} shadow-2 z-1` : `bg-${offColor}`}`}
            disabled={disabled ? 'disabled' : null}
            onClick={onClick}
          >
            <span className={`${isOn ? 'gray' : 'white'} f7 tracked-light ttu`}>
              { showText ? 'off' : null }
            </span>
          </button>
          <button
            className={`bn w-50
              ${disabled ? 'o-60' : 'pointer'}
              ${isOn ? `bg-${onColor} gray` : `bg-${baseColor} shadow-2`}`}
            disabled={disabled ? 'disabled' : null}
            onClick={onClick}
          >
            <span className={`f7 tracked-light ttu ${isOn ? 'white' : 'gray'}`}>
              { showText ? 'on' : null}
            </span>
          </button>
        </div>
      </div>
    );
  }
}

SplitButton.propTypes = {
  baseColor: PropTypes.string,
  disabled: PropTypes.bool,
  isOn: PropTypes.bool,
  offColor: PropTypes.string,
  onClick: PropTypes.func,
  onColor: PropTypes.string,
  pill: PropTypes.bool,
  radius: PropTypes.number,
  showText: PropTypes.bool,
};

SplitButton.defaultProps = {
  baseColor: 'moon-gray',
  disabled: false,
  isOn: false,
  onClick: null,
  offColor: 'dark-red',
  offText: 'off',
  onColor: 'green',
  onText: 'on',
  pill: false,
  radius: 0,
  showText: true,
};

export default SplitButton;
