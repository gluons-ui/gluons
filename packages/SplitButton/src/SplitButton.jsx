import React from 'react';
import PropTypes from 'prop-types';
import clamp from 'lodash/clamp';

const SplitButton = ({
  baseColor,
  disabled,
  isOn,
  offColor,
  onClick,
  onColor,
  pill,
  radius,
  showText }) => {
  const isDisabled = `${disabled ? 'o-60' : 'pointer'}`;
  return (
    <div className="split-button">
      <div
        className={`b--gray ba
            ${pill ? 'br-pill' : `br${clamp(radius, 0, 3)}`} flex h2 overflow-hidden w4`}
      >
        <button
          className={`bn w-50
              ${isDisabled}
              ${isOn ? `bg-${baseColor} shadow-2 z-1` : `bg-${offColor}`}`}
          disabled={disabled ? 'disabled' : null}
          onClick={onClick}
        >
          <span className={`f7 tracked-light ttu ${isOn ? 'gray' : 'white'} `}>
            { showText ? 'off' : null }
          </span>
        </button>
        <button
          className={`bn w-50
              ${isDisabled}
              ${isOn ? `bg-${onColor} gray` : `bg-${baseColor} shadow-2`}`}
          disabled={disabled ? 'disabled' : null}
          onClick={onClick}
        >
          <span className={`f7 tracked-light ttu ${isOn ? 'white' : 'gray'}`}>
            { showText ? 'on' : null}
          </span>
        </button>
      </div>
    </div>
  );
};


SplitButton.propTypes = {
  baseColor: PropTypes.string,
  disabled: PropTypes.bool,
  isOn: PropTypes.bool,
  offColor: PropTypes.string,
  onClick: PropTypes.func.isRequired,
  onColor: PropTypes.string,
  pill: PropTypes.bool,
  radius: PropTypes.number,
  showText: PropTypes.bool,
};

SplitButton.defaultProps = {
  baseColor: 'moon-gray',
  disabled: false,
  isOn: false,
  offColor: 'dark-red',
  offText: 'off',
  onColor: 'green',
  onText: 'on',
  pill: false,
  radius: 0,
  showText: true,
};

export default SplitButton;
