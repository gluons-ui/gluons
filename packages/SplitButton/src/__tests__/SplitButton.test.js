import React from 'react';
import renderer from 'react-test-renderer';
import SplitButton from '../SplitButton';

describe('<SplitButton />', () => {
  it('renders Off correctly', () => {
    const comp = renderer.create(
      <SplitButton />,
    ).toJSON();
    expect(comp).toMatchSnapshot();
  });
  it('renders On correctly', () => {
    const comp = renderer.create(
      <SplitButton
        isOn
      />,
    ).toJSON();
    expect(comp).toMatchSnapshot();
  });
});
