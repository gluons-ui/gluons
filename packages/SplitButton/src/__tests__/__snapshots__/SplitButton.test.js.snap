// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SplitButton /> renders Off correctly 1`] = `
<div
  className="split-button"
>
  <div
    className="b--gray ba
              br0 flex h2 overflow-hidden w4"
  >
    <button
      className="bn w-50
                  pointer
                  bg-dark-red"
      disabled={null}
      onClick={null}
    >
      <span
        className="white f7 tracked-light ttu"
      >
        off
      </span>
    </button>
    <button
      className="bn w-50
                  pointer
                  bg-moon-gray shadow-2"
      disabled={null}
      onClick={null}
    >
      <span
        className="f7 tracked-light ttu gray"
      >
        on
      </span>
    </button>
  </div>
</div>
`;

exports[`<SplitButton /> renders On correctly 1`] = `
<div
  className="split-button"
>
  <div
    className="b--gray ba
              br0 flex h2 overflow-hidden w4"
  >
    <button
      className="bn w-50
                  pointer
                  bg-moon-gray shadow-2 z-1"
      disabled={null}
      onClick={null}
    >
      <span
        className="gray f7 tracked-light ttu"
      >
        off
      </span>
    </button>
    <button
      className="bn w-50
                  pointer
                  bg-green gray"
      disabled={null}
      onClick={null}
    >
      <span
        className="f7 tracked-light ttu white"
      >
        on
      </span>
    </button>
  </div>
</div>
`;
